lfs/client.go

```diff
@@ -259,11 +287,21 @@ func UploadCheck(oidPath string) (*objectResource, error) {
 
 	tracerx.Printf("api: uploading (%s)", oid)
 	res, obj, err := doLegacyApiRequest(req)
+	
+	io.Copy(ioutil.Discard, res.Body)
+	res.Body.Close()
+	
 	if err != nil {
 		if IsAuthError(err) {
-			Config.SetAccess("basic")
-			tracerx.Printf("api: upload check not authorized, submitting with auth")
-			return UploadCheck(oidPath)
+			if strings.ToLower(res.Header["Www-Authenticate"][0][0:4]) == "ntlm" {
```

Comment:
This looks like it could lead to panics. I'd recommend doing the following:

* Use `res.Header.Get("Www-Authenticate")`, which always returns a string.
* Use `strings.HasPrefix(wwwAuth, "ntlm")`, which won't panic if the header value only has 3 characters.
* Wrap this in a small helper func that takes an `*http.Response`. Looks like this check is repeated in multiple places.

```golang
authType := getAuthType(res)
Config.SetAccess(authType)
tracerx.Printf("api: upload check not authorized, submitting with %s auth", authType)
return UploadCheck(oidPath)
```